src/filtering.jl

```diff
@@ -0,0 +1,128 @@
+## Various methods for filtering and sorting Documentation
+
+
+"""
+Filter Documentation based on categories or file source
+
+```julia
+Base.filter(docs::Documentation; categories = Symbol[], files = String[])
+```
+
+**Arguments**
+
+* `docs` : main input
+
+**Optional keyword arguments**
+
+* `categories` : categories to include in the result; can include any
+  of [:module, :function, :method, :type, :macro, :global]
+
+* `files` : vector of file names where entries originated to include
+  in the result; full pathnames are searched; can include partial
+  paths
+
+**Returns**
+
+* `::Documentation` : the filtered result
+
+**Example**
+
+```julia
+using Lexicon, Docile, Docile.Interface
+d = documentation(Docile);
+
+# Filter entries with categories of :macro and :type
+entries( filter(d, categories = [:macro, :type]) )
+
+# Filter entries from the file types.jl
+entries( filter(d, files = ["types.jl"]) )
+```
+
+"""
+function Base.filter(docs::Documentation; categories = Symbol[], files = String[])
```

Comment:
`Documentation` is now just an alias for `Metadata`, so here and further down they can be replaced with `Metadata` for consistency.

See [interface.jl](https://github.com/MichaelHatherly/Docile.jl/blob/6a7d5513f229c3068d778886577b2b15579c70a7/src/interface.jl#L21). "Documentation" seemed like the incorrect word to be using since it isn't just docstrings being store in the struct anymore.